samtranslator/model/sam_resources.py

```diff
@@ -2163,7 +2170,63 @@ def _construct_appsync_api(self) -> GraphQLApi:
         if self.Tags:
             api.Tags = get_tag_list(self.Tags)
 
-        return api
+        # Logging has 3 possible types: dict, bool, and None.
+        # GraphQLApi will not include logging if and only if the user explicity sets Logging as false boolean.
+        # It will for every other value (including true boolean which is essentially same as None).
+        if isinstance(self.Logging, bool) and self.Logging is False:
+            return api, None
+
+        api.LogConfig, cloudwatch_role = (
+            self._parse_logging_properties() if isinstance(self.Logging, dict) else self._create_logging_default()
+        )
+
+        return api, cloudwatch_role
+
+    def _create_logging_default(self) -> Tuple[LogConfigType, IAMRole]:
+        """
+        Create a default logging configuration.
+
+        This function should only be called when "Logging" property is a False boolean or NoneType.
+        """
+        log_config: LogConfigType = {}
+        log_config["FieldLogLevel"] = "ALL"
+        cloudwatch_role = self._construct_cloudwatch_role()
+        log_config["CloudWatchLogsRoleArn"] = cloudwatch_role.get_runtime_attr("arn")
+
+        return log_config, cloudwatch_role
+
+    def _parse_logging_properties(self) -> Tuple[LogConfigType, Optional[IAMRole]]:
+        """
+        Parse logging properties from SAM template, and use defaults if required keys dont exist.
+
+        This function should only be called when "Logging" property is a dictionary type.
+        """
+        logging = cast(Dict[str, Any], self.Logging)
+        log_config: LogConfigType = {}
+
+        log_config["FieldLogLevel"] = logging["FieldLogLevel"] if "FieldLogLevel" in logging else "ALL"
+        if "ExcludeVerboseContent" in logging:
+            log_config["ExcludeVerboseContent"] = logging["ExcludeVerboseContent"]
+
+        cloudwatch_role = self._construct_cloudwatch_role() if "CloudWatchLogsRoleArn" not in logging else None
+        log_config["CloudWatchLogsRoleArn"] = (
+            cloudwatch_role.get_runtime_attr("arn") if cloudwatch_role else logging["CloudWatchLogsRoleArn"]
+        )
+        return log_config, cloudwatch_role
```

Comment:
Why return `cloudwatch_role` separately if it's already part of `LogConfigType`?

Reply:
cloudwatch_role is an IAMRole resource so we can create it, the field in `LogConfigType` is just the arn